usaspending_api/disaster/v2/views/agency/count.py

```diff
@@ -21,36 +18,10 @@ class AgencyCountViewSet(AwardTypeMixin, FabaOutlayMixin, DisasterBase):
 
     @cache_response()
     def post(self, request: Request) -> Response:
-        filters = [
-            self.all_closed_defc_submissions,
-            self.has_award_of_provided_type(),
-            self.is_in_provided_def_codes,
-        ]
-
-        if self.award_type_codes:
-            count = (
-                CovidFinancialAccountMatview.objects.annotate(cast_def_codes=Cast("def_codes", ArrayField(TextField())))
-                .filter(
-                    type__in=self.award_type_codes,
-                    cast_def_codes__overlap=self.def_codes,
-                    award__funding_agency__toptier_agency__isnull=False,
-                )
-                .values("award__funding_agency__toptier_agency")
-                .distinct()
-                .count()
-            )
-
-        else:
-            filters.extend(
-                [Q(treasury_account__funding_toptier_agency=OuterRef("pk")), self.is_non_zero_total_spending]
-            )
-            count = (
-                ToptierAgency.objects.annotate(
-                    include=Exists(FinancialAccountsByProgramActivityObjectClass.objects.filter(*filters).values("pk"))
-                )
-                .filter(include=True)
-                .values("pk")
-                .count()
-            )
-
-        return Response({"count": count})
+        filter_query = QueryWithFilters.generate_awards_elasticsearch_query(self.filters)
+        search = AwardSearch().filter(filter_query)
+        search.update_from_dict({"size": 0})
+        search.aggs.bucket("agency_count", create_count_aggregation("funding_toptier_agency_agg_key"))
```

Comment:
Running into performance issues with this one as well against QAT. Changing to use the `.hash` field seems to help. Not sure why the agg_key fields are proving to be so problematic.